plugin/src/main/java/io/jenkins/plugins/remotingkafka/GlobalKafkaConfiguration.java

```diff
@@ -21,33 +45,288 @@ public static GlobalKafkaConfiguration get() {
         return GlobalConfiguration.all().get(GlobalKafkaConfiguration.class);
     }
 
-    public String getConnectionURL() {
-        return connectionURL;
+    public String getBrokerURL() {
+        return brokerURL;
     }
 
     public String getZookeeperURL() {
         return zookeeperURL;
     }
 
-    public FormValidation doCheckConnectionURL(@QueryParameter String connectionURL) {
-        if (StringUtils.isBlank(connectionURL)) {
-            return FormValidation.warning(Messages.GlobalKafkaConfiguration_KafkaConnectionURLWarning());
+    public boolean getEnableSSL() {
+        return enableSSL;
+    }
+
+    public String getKafkaCredentialsId() {
+        return kafkaCredentialsId;
+    }
+
+    public String getSslTruststoreCredentialsId() {
+        return sslTruststoreCredentialsId;
+    }
+
+    public String getSslKeystoreCredentialsId() {
+        return sslKeystoreCredentialsId;
+    }
+
+    public String getSslKeyCredentialsId() {
+        return sslKeyCredentialsId;
+    }
+
+    public String getKafkaUsername() {
+        return getCredential(kafkaCredentialsId).getUsername();
+    }
+
+    public String getKafkaPassword() {
+        return getCredential(kafkaCredentialsId).getPassword().getPlainText();
+    }
+
+    public String getSSLTruststoreLocation() {
+        return getCredential(sslTruststoreCredentialsId).getUsername();
+    }
+
+    public String getSSLTruststorePassword() {
+        return getCredential(sslTruststoreCredentialsId).getPassword().getPlainText();
+    }
+
+    public String getSSLKeystoreLocation() {
+        return getCredential(sslKeystoreCredentialsId).getUsername();
+    }
+
+    public String getSSLKeystorePassword() {
+        return getCredential(sslKeystoreCredentialsId).getPassword().getPlainText();
+    }
+
+    public String getSSLKeyPassword() {
+        return getCredential(sslKeyCredentialsId).getPassword().getPlainText();
+    }
+
+    public FormValidation doCheckBrokerURL(@QueryParameter("brokerURL") final String brokerURL) {
+        if (StringUtils.isBlank(brokerURL)) {
+            return FormValidation.error(Messages.GlobalKafkaConfiguration_KafkaConnectionURLWarning());
         }
         return FormValidation.ok();
     }
 
-    public FormValidation doCheckZookeeperURL(@QueryParameter String zookeeperURL) {
+    public FormValidation doCheckZookeeperURL(@QueryParameter("zookeeperURL") String zookeeperURL) {
         if (StringUtils.isBlank(zookeeperURL)) {
-            return FormValidation.warning(Messages.GlobalKafkaConfiguration_ZookeeperURLWarning());
+            return FormValidation.error(Messages.GlobalKafkaConfiguration_ZookeeperURLWarning());
         }
         return FormValidation.ok();
     }
 
+    public FormValidation doTestZookeeperConnection(@QueryParameter("zookeeperURL") final String zookeeperURL)
```

Comment:
I suggest securing these methods by adding `@RequirePOST` and by checking `Jenkins.ADMINISTER` permissions in the call